.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,30 @@
+name: NodeJS
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 13.x]
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Cache Node Modules
+        uses: actions/cache@v1
+        id: cache
+        with:
+          path: node_modules
+          key: ${{ runner.OS }}-node-modules-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Install NPM
+        if: steps.cache.outputs.cache-hit != 'true'
```

Comment:
It seems to me that build and test won't happen if node modules are still cached? Or am I wrong?

Reply:
Agreed.  There needs to be another, separate step after `Install NPM` that always runs `build` and `test`

Reply:
I have added these extra steps